optimum/habana/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py

```diff
@@ -150,222 +291,245 @@ def gaudi_gpt_bigcode_block_forward(
 
     return outputs  # hidden_states, present, (attentions, cross_attentions)
 
-
-def gaudi_gpt_bigcode_model_forward(
-    self,
-    input_ids: Optional[torch.Tensor] = None,
-    past_key_values: Optional[List[torch.Tensor]] = None,
-    attention_mask: Optional[torch.Tensor] = None,
-    token_type_ids: Optional[torch.Tensor] = None,
-    position_ids: Optional[torch.Tensor] = None,
-    head_mask: Optional[torch.Tensor] = None,
-    inputs_embeds: Optional[torch.Tensor] = None,
-    encoder_hidden_states: Optional[torch.Tensor] = None,
-    encoder_attention_mask: Optional[torch.Tensor] = None,
-    use_cache: Optional[bool] = None,
-    output_attentions: Optional[bool] = None,
-    output_hidden_states: Optional[bool] = None,
-    return_dict: Optional[bool] = None,
-    token_idx: Optional[torch.Tensor] = None,
-) -> Union[Tuple, BaseModelOutputWithPastAndCrossAttentions]:
+class GaudiGPTBigCodeModel(GPTBigCodeModel):
     """
-    Copied from GPTBigCodeModel.forward: https://github.com/huggingface/transformers/blob/main/src/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py
+    Inherits from GPTBigCodeModel: https://github.com/huggingface/transformers/blob/v4.40-release/src/transformers/models/gpt_bigcode/modeling_gpt_bigcode.py
```

Comment:
Assuming this is v4.40.2 ? Might be better to be specific in case the branch moves on after we commit this.

Reply:
Last commit with transformers package version bump for this model is to ver 4.40, see [this](https://github.com/huggingface/optimum-habana/pull/1027) PR. What do you mean by "specific" in case of this link?

Reply:
you may just post the link you want to see here and I will use it

Reply:
Sorry ignore my comment I see we have `"transformers >= 4.40.0, < 4.41.0"`, so using the v4.40 release branch is fine